src/main/java/com/amuzil/omegasource/magus/skill/util/traits/skilltraits/XPTrait.java

```diff
@@ -0,0 +1,42 @@
+package com.amuzil.omegasource.magus.skill.util.traits.skilltraits;
+
+import com.amuzil.omegasource.magus.skill.util.traits.SkillTrait;
+import net.minecraft.nbt.CompoundTag;
+
+public class XPTrait extends SkillTrait {
+
+    private double xp;
+
+    public XPTrait(double xp, String name) {
+        super(name);
+        this.xp = xp;
+    }
+
+    @Override
+    public CompoundTag serializeNBT() {
+        CompoundTag tag = super.serializeNBT();
+        tag.putDouble(getName(), xp);
+        return tag;
+    }
+
+    @Override
+    public void deserializeNBT(CompoundTag nbt) {
+        super.deserializeNBT(nbt);
+        xp = nbt.getDouble(getName());
+    }
+
+    public void setXp(double xp) {
```

Comment:
```suggestion
    public void setXP(double xp) {
```

Be consistent with the class naming

Reply:
My bad